### PR TITLE
Amogus generator

### PR DESCRIPTION
**Pull request by PaperDoggy**

### CODE REVIEW

**Comment by pixelpapercraft** [comment]
Great work @PaperDoggy 👍 

Both of these generators are candidates for publishing live.

There are a few steps needed to get them published, I'll add some comments for you.

Also, in future it's a good idea to create a separate PR for each generator (this helps to get them released more quickly).

File: src/generators/bendable-amogus/BendableAmogus.res [hunk 1]
```diff
@@ -0,0 +1,239 @@
+let requireImage = fileName => Generator.requireImage("./images/" ++ fileName)
+let requireTexture = fileName => Generator.requireImage("./textures/" ++ fileName)
+
+let id = "bendableamogusgenerator"
+
+let name = "Bendable Amogus generator"
```
Comment (pixelpapercraft) [inline src/generators/bendable-amogus/BendableAmogus.res:6]: Let's rename this generator:

Directory name: `among-us-bendable`

File name: `AmongUsBendableGenerator.res`

id: `among-us-bendable`

name: `Among Us Bendable`

File: src/generators/bendable-amogus/BendableAmogus.res [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+        
+      ]);
+
+      Generator.defineTextureInput("Skin", {
+        standardWidth: 64,
+        standardHeight: 64,
+        choices: []
+      });
+      Generator.drawImage("Background", (0, 0))
+  let red = Generator.getSelectInputValue("Color") === "Red";
```
Comment (pixelpapercraft) [inline src/generators/bendable-amogus/BendableAmogus.res:58]: There are a few formatting problems in the file.

If you use VSCode for editing the files then install the `rescript-vscode` extension. This should auto-format the file each time you save the file.

File: src/generators/bendable-amogus/BendableAmogus.res [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+
+
+  
+if red {
+  Generator.drawTextureLegacy(
+    "Colors",
+    {x: 0, y: 0, w: 1, h: 1},
+    {x: 0, y: 0, w: 600, h: 660},
+    (),
+  ) };
```
Comment (pixelpapercraft) [inline src/generators/bendable-amogus/BendableAmogus.res:85]: If you like, there is a slightly less verbose syntax you can use here for the rectangles.

Instead of the curly bracket syntax `{x: 0, y: 0, w: 1, h: 1}`, you can use the round bracket syntax `(0, 0, 1, 1)`.

So this will look like:


```
Generator.drawTexture(
  "Colors",
  (0, 0, 1, 1),
  (0, 0, 600, 660),
  ()
)
```

Whichever you prefer is fine.

File: src/generators/bendable-amogus/BendableAmogus.res [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+ Generator.drawImage("Folds", (0, 0))
+
+
+  
+}
+
+let generator: Generator.generatorDef = {
+  id: id,
+  name: name,
+  thumbnail: None,
```
Comment (pixelpapercraft) [inline src/generators/bendable-amogus/BendableAmogus.res:233]: Please add a thumbnail - it should be a 256 x 256 JPG image.

File: src/generators/minecraft-mutant-character/MutantCharacter.res [hunk 5]
```diff
@@ -0,0 +1,507 @@
+let requireImage = fileName => Generator.requireImage("./images/" ++ fileName)
+let requireTexture = fileName => Generator.requireImage("./textures/" ++ fileName)
+
+let id = "mutantgenerator"
+
+let name = "Mutant Character"
```
Comment (pixelpapercraft) [inline src/generators/minecraft-mutant-character/MutantCharacter.res:6]: Let's rename this generator:

Directory name: `minecraft-mutant-character`

File name: `MinecraftMutantCharacterGenerator.res`

id: `minecraft-mutant-character`

name: `Mutant Character`

File: src/generators/minecraft-mutant-character/MutantCharacter.res [hunk 6]
```diff
@@ -0,0 +1,507 @@
...
+let video: Generator.videoDef = {url: "https://www.youtube.com/embed/DVzumgRinjY"}
```
Comment (pixelpapercraft) [inline src/generators/minecraft-mutant-character/MutantCharacter.res:7]: Nice work with the video

File: src/generators/minecraft-mutant-character/MutantCharacter.res [hunk 7]
```diff
@@ -0,0 +1,507 @@
...
+  url: Generator.requireImage("./thumbnail/thumbnail.jpeg"),
+}
+let images: array<Generator.imageDef> = [
+  {id: "Page1", url: requireImage("Page1.png")},
+  {id: "Page2", url: requireImage("Page2.png")},
+  {id: "Page3", url: requireImage("Page3.png")},
+  {id: "Page4", url: requireImage("Page4.png")},
+  {id: "Hole1", url: requireImage("Hole1.png")},
+  {id: "hole2", url: requireImage("hole2.png")},
+  {id: "hole3", url: requireImage("hole3.png")},
```
Comment (pixelpapercraft) [inline src/generators/minecraft-mutant-character/MutantCharacter.res:18]: For consistency, would be good to rename `hole2` and `hole3` to `Hole2` and `Hole3`

File: src/generators/minecraft-mutant-character/MutantCharacter.res [hunk 8]
```diff
@@ -0,0 +1,507 @@
...
+let script = () => {
+  
+Generator.defineTextureInput("Skin", {
+        standardWidth: 64,
+        standardHeight: 64,
+        choices: []
+      });
+
+
+Generator.defineSelectInput("Color", [
```
Comment (pixelpapercraft) [inline src/generators/minecraft-mutant-character/MutantCharacter.res:40]: Maybe rename to `Skin style` or something like that?

File: src/generators/minecraft-mutant-character/MutantCharacter.res [hunk 9]
```diff
@@ -0,0 +1,507 @@
...
+        "Steve",
+        "Alex",
+        
+        
+      ]);
+let alexModel = Generator.getSelectInputValue("Color") === "Alex";
```
Comment (pixelpapercraft) [inline src/generators/minecraft-mutant-character/MutantCharacter.res:46]: Nice work creating the Alex version.

File: src/generators/minecraft-mutant-character/MutantCharacter.res [hunk 10]
```diff
@@ -0,0 +1,507 @@
...
+    { x: 48, y: 0, w: 8, h: 8 },
+    { x: 2, y: 174, w: 84, h: 84 },
+    ~flip=#Vertical,
+  (),
+  ); // Bottom
+  
+  ////Right Arm
+  
+  if (alexModel) {
+      Generator.drawTextureLegacy( "Skin", { x: 40, y: 20, w: 4, h: 6  }, {x:200, y:330, w:64, h:168},
```
Comment (pixelpapercraft) [inline src/generators/minecraft-mutant-character/MutantCharacter.res:124]: Formatting needs to be fixed up. As mentioned above, install the `rescript-vscode` extension inside VSCode and it should format the files correctly for you.

**Comment by pixelpapercraft** [comment]
As you work through the comments, please reply to each comment where I've requested a change and let me know how you go.

Good work.

File: src/generators/bendable-amogus/BendableAmogus.res (hunk 2 above)
Comment (PaperDoggy) [reply src/generators/bendable-amogus/BendableAmogus.res:60]: I already have it installed, but it does not changes anything. How can I fix it?

File: src/generators/bendable-amogus/BendableAmogus.res (hunk 4 above)
Comment (PaperDoggy) [reply src/generators/bendable-amogus/BendableAmogus.res]: Done

File: src/generators/minecraft-mutant-character/MutantCharacter.res (hunk 8 above)
Comment (PaperDoggy) [reply src/generators/minecraft-mutant-character/MutantCharacter.res:40]: Fixed that(I was using Amogus generator as base so :) )

File: src/generators/minecraft-mutant-character/MutantCharacter.res (hunk 5 above)
Comment (PaperDoggy) [reply src/generators/minecraft-mutant-character/MutantCharacter.res:6]: Done

File: src/generators/minecraft-mutant-character/MutantCharacter.res (hunk 7 above)
Comment (PaperDoggy) [reply src/generators/minecraft-mutant-character/MutantCharacter.res:18]: Done

File: src/generators/minecraft-mutant-character/MutantCharacter.res (hunk 10 above)
Comment (PaperDoggy) [reply src/generators/minecraft-mutant-character/MutantCharacter.res:124]: Done

File: src/generators/bendable-amogus/BendableAmogus.res (hunk 2 above)
Comment (PaperDoggy) [reply src/generators/bendable-amogus/BendableAmogus.res:60]: Done

File: src/generators/Generators.res [hunk 11]
```diff
@@ -36,9 +36,10 @@ let utility: array<Generator.generatorDef> = [
   MinecraftCharacterHeadsGenerator.generator,
 ]
 
-let mod: array<Generator.generatorDef> = [DalekModDalekGenerator.generator]
+let mod: array<Generator.generatorDef> = [DalekModDalekGenerator.generator,
+MinecraftMutantCharacterGenerator.generator]
```
Comment (pixelpapercraft) [inline src/generators/Generators.res:40]: Need a formatting fix here

File: src/generators/bendable-amogus/AmongUsBendableGenerator.res [hunk 12]
```diff
@@ -0,0 +1,254 @@
+let requireImage = fileName => Generator.requireImage("./images/" ++ fileName)
+let requireTexture = fileName => Generator.requireImage("./textures/" ++ fileName)
+
+let id = "bendableamogusgenerator"
+
```
Comment (pixelpapercraft) [inline src/generators/bendable-amogus/AmongUsBendableGenerator.res:5]: Let's rename the **directory** from `bendable-amongus` to `among-us-bendable`

File: src/generators/bendable-amogus/AmongUsBendableGenerator.res [hunk 13]
```diff
@@ -0,0 +1,254 @@
+let requireImage = fileName => Generator.requireImage("./images/" ++ fileName)
+let requireTexture = fileName => Generator.requireImage("./textures/" ++ fileName)
+
+let id = "bendableamogusgenerator"
```
Comment (pixelpapercraft) [inline src/generators/bendable-amogus/AmongUsBendableGenerator.res:4]: Let's rename the `id` to `among-us-bendable`

File: src/generators/bendable-amogus/AmongUsBendableGenerator.res [hunk 14]
```diff
@@ -0,0 +1,254 @@
...
+
+let name = "Bendable Amogus generator"
```
Comment (pixelpapercraft) [inline src/generators/bendable-amogus/AmongUsBendableGenerator.res:6]: Let's make the name `Among Us Bendable` (no need for the word "generator" here)

File: src/generators/bendable-amogus/AmongUsBendableGenerator.res [hunk 15]
```diff
@@ -0,0 +1,254 @@
...
+let thumbnail: Generator.thumnbnailDef = {
+  url: Generator.requireImage("./thumbnails/thumbnail.jpeg"),
```
Comment (pixelpapercraft) [inline src/generators/bendable-amogus/AmongUsBendableGenerator.res:8]: For consistency with other generators let's rename `thumbnails` to `thumbnail`

File: src/generators/bendable-amogus/AmongUsBendableGenerator.res [hunk 16]
```diff
@@ -0,0 +1,254 @@
...
+    ~rotateLegacy=0.0,
+    (),
+  )
+  Generator.drawImage("Folds", (0, 0))
+}
+
+let generator: Generator.generatorDef = {
+  id: id,
+  name: name,
+  thumbnail: Some(thumbnail),
```
Comment (pixelpapercraft) [inline src/generators/bendable-amogus/AmongUsBendableGenerator.res:248]: Good work on the thumbnail

File: src/generators/bendable-amogus/BendableAmogus.res [hunk 17]
```diff
@@ -0,0 +1,241 @@
+let requireImage = fileName => Generator.requireImage("./images/" ++ fileName)
+let requireTexture = fileName => Generator.requireImage("./textures/" ++ fileName)
+
+let id = "bendableamogusgenerator"
```
Comment (pixelpapercraft) [inline src/generators/bendable-amogus/BendableAmogus.res:4]: We can delete this old file now.

File: src/generators/biome/BiomeGenerator.res [hunk 18]
```diff
@@ -0,0 +1,182 @@
+let requireImage = fileName => Generator.requireImage("./images/" ++ fileName)
+let requireTexture = fileName => Generator.requireImage("./textures/" ++ fileName)
+
+let id = "biome"
```
Comment (pixelpapercraft) [inline src/generators/biome/BiomeGenerator.res:4]: Let's leave the Biome generator for another PR and delete from this PR.

File: src/generators/minecraft-mutant-character/MutantCharacter.res [hunk 19]
```diff
@@ -0,0 +1,507 @@
+let requireImage = fileName => Generator.requireImage("./images/" ++ fileName)
+let requireTexture = fileName => Generator.requireImage("./textures/" ++ fileName)
+
+let id = "mutantgenerator"
```
Comment (pixelpapercraft) [inline src/generators/minecraft-mutant-character/MutantCharacter.res:4]: Let's delete this old file

File: src/generators/minecraft-mutant-character/MinecraftMutantCharacterGenerator.res [hunk 20]
```diff
@@ -0,0 +1,507 @@
...
+  }
+]
+
+let script = () => {
+  
+Generator.defineTextureInput("Skin", {
+        standardWidth: 64,
+        standardHeight: 64,
+        choices: []
+      });
```
Comment (pixelpapercraft) [inline src/generators/minecraft-mutant-character/MinecraftMutantCharacterGenerator.res:37]: Formatting needs to be fixed in this file

Comment (PaperDoggy) [reply src/generators/minecraft-mutant-character/MinecraftMutantCharacterGenerator.res]: Done

File: src/generators/minecraft-mutant-character/MutantCharacter.res (hunk 19 above)
Comment (PaperDoggy) [reply src/generators/minecraft-mutant-character/MutantCharacter.res]: Done

File: src/generators/biome/BiomeGenerator.res (hunk 18 above)
Comment (PaperDoggy) [reply src/generators/biome/BiomeGenerator.res]: Done

File: src/generators/bendable-amogus/BendableAmogus.res (hunk 17 above)
Comment (PaperDoggy) [reply src/generators/bendable-amogus/BendableAmogus.res]: Done

File: src/generators/bendable-amogus/AmongUsBendableGenerator.res (hunk 15 above)
Comment (PaperDoggy) [reply src/generators/bendable-amogus/AmongUsBendableGenerator.res]: Done

File: src/generators/bendable-amogus/AmongUsBendableGenerator.res (hunk 14 above)
Comment (PaperDoggy) [reply src/generators/bendable-amogus/AmongUsBendableGenerator.res]: Done

File: src/generators/bendable-amogus/AmongUsBendableGenerator.res (hunk 13 above)
Comment (PaperDoggy) [reply src/generators/bendable-amogus/AmongUsBendableGenerator.res]: Done

File: src/generators/bendable-amogus/AmongUsBendableGenerator.res (hunk 12 above)
Comment (PaperDoggy) [reply src/generators/bendable-amogus/AmongUsBendableGenerator.res]: Done

File: src/generators/Generators.res (hunk 11 above)
Comment (PaperDoggy) [reply src/generators/Generators.res]: Done

File: src/generators/minecraft-mutant-character/MinecraftMutantCharacterGenerator.res [hunk 21]
```diff
@@ -0,0 +1,993 @@
...
+    "Skin",
+    {
+      standardWidth: 64,
+      standardHeight: 64,
+      choices: [],
+    },
+  )
+
+  Generator.defineSelectInput("Skin style", ["Steve", "Alex"])
+  let alexModel = Generator.getSelectInputValue("Color") === "Alex"
```
Comment (pixelpapercraft) [inline src/generators/minecraft-mutant-character/MinecraftMutantCharacterGenerator.res:41]: Does `"Color"` here need to be changed to `"Skin style"`?

**Comment by pixelpapercraft** [comment]
Let's remove these two files (the lower case versions):

```
src/generators/minecraft-mutant-character/images/hole2.png
src/generators/minecraft-mutant-character/images/hole3.png
```

And only keep the upper case versions of the name.

File: src/generators/minecraft-mutant-character/MinecraftMutantCharacterGenerator.res (hunk 21 above)
Comment (pixelpapercraft) [reply src/generators/minecraft-mutant-character/MinecraftMutantCharacterGenerator.res]: Hi @PaperDoggy just a reminder about this fix ^^^

File: src/generators/Generators.res [hunk 22]
```diff
@@ -36,9 +36,12 @@ let utility: array<Generator.generatorDef> = [
   MinecraftCharacterHeadsGenerator.generator,
 ]
 
-let mod: array<Generator.generatorDef> = [DalekModDalekGenerator.generator]
+let mod: array<Generator.generatorDef> = [
+  DalekModDalekGenerator.generator,
+  MinecraftMutantCharacterGenerator.generator,
```
Comment (pixelpapercraft) [inline src/generators/Generators.res:41]: Let's put your Mutant Generator first and the Dalek generator second here.

File: src/generators/minecraft-mutant-character/MinecraftMutantCharacterGenerator.res (hunk 21 above)
Comment (pixelpapercraft) [reply src/generators/minecraft-mutant-character/MinecraftMutantCharacterGenerator.res]: Nevermind, just saw you have fixed this 👍

File: src/generators/Generators.res (hunk 22 above)
Comment (pixelpapercraft) [reply src/generators/Generators.res:41]: Actually, I'll take care for that for you

**Comment by pixelpapercraft** [comment]
Good work @PaperDoggy this is good to merge now.
